src/python/WMCore/Services/MSUtils.py

```diff
@@ -0,0 +1,31 @@
+"""
+This module contains a few basic WMCore-related utilitarian functions
+"""
+
+from Utils.CertTools import ckey, cert
+
+from WMCore.Services.pycurl_manager import RequestHandler
+
+
+def makeHttpRequest(srvUrl, payload=None, method='GET'):
+    """
+    Fetch documents from given service URL according to the query dictionary
+
+    :param srvUrl: string with the service url
+    :param payload: payload query or document, in case of GET HTTP request it is query dictionary,
+        in case of POST/PUT HTTP requests it is JSON payload to the service
+    :param method: string, defines which HTTP method to use
+    :return: returns a list with all objects, or raises
+        an exception in case of failure
+    """
+    payload = payload or {}
+    mgr = RequestHandler()
+    headers = {'Content-Type': 'application/json'}
+    data = mgr.getdata(srvUrl, payload, headers, verb=method,
+                       ckey=ckey(), cert=cert(), encode=True, decode=True)
+    if data and data.get("result", []):
+        if "error" in data["result"][0]:
+            msg = f"Failed to contact {srvUrl} via {method} request with {payload}"
+            msg += f" and error message: {data}"
+            raise RuntimeError(msg)
```

Comment:
I think it would be better here to raise either  directly a specific HTTP error (kind of speculative, because we cannot be sure if exactly HTTP is always to blame for an empty `data`/result from the call to MSTransferror - it would require more elaborate method for HTTP error handling plugged in here - maybe too much though) or a dedicated exception just for this communication here only, or maybe in the worst case a generic WMCore exception, but the later won't give us much in terms of error discrimination.

Reply:
Again, I don't have strong preferences here since python exceptions are not the best mechanism to handle HTTP issues.